src/main/java/com/internetitem/logback/elasticsearch/AbstractElasticsearchPublisher.java

```diff
@@ -119,12 +119,16 @@ public void run() {
 			try {
 				Thread.sleep(settings.getSleepTime());
 
+				if (currentTry > maxRetries) {
```

Comment:
I wonder if we should handle retries differently. I.e. if one update action failed, the retry is tried again (without taking any new events into consideration). That would cause less events to be dropped in case the maximum retry counter is reached.
On the other side, this may block further writes for quite some time (depending on the sleep time). Maybe we could introduce a thread-pool that is working on that queue of events (which would still cause events to be written in case one of the thread is busy with retrying)

Reply:
WDYT @FKorni2 ?

Reply:
Sounds legit to not add more elements when we retry. I am not so sure if it is a good idea to have another thread to push other elements in the meanwhile as we don't know what the current issue is. e.g. if the cluster is overloaded we would just spam it with more requests.